src/solution.c

```diff
@@ -468,12 +472,49 @@ static void solution_thread(void *arg)
       continue;
     }
 
+    /* check if we have a solution, if yes calc iono correction */
+    if (soln_flag) {
+      for (u8 i = 0; i < n_ready_tdcp; i++) {
+        ionosphere_t iono_params;
+        double az, el, iono_correction, tropo_correction, cp_correction;
+        /* read iono parameters for the tracking channel if available */
+        if (gps_iono_params_read(&iono_params)) {
+          wgsecef2azel(nav_meas_tdcp[i].sat_pos, position_solution.pos_ecef,
+                       &az, &el);
+          /* calculate iono correction */
+          iono_correction = calc_ionosphere(&nav_meas_tdcp[i].tot,
+                                            position_solution.pos_llh[0],
+                                            position_solution.pos_llh[1],
+                                            az,el,
+                                            &iono_params);
+          /* calculate tropo correction */
+          tropo_correction = calc_troposphere(&nav_meas_tdcp[i].tot,
+                                              position_solution.pos_llh[0],
+                                              position_solution.pos_llh[2],
```

Comment:
I think there should also be a note that we are currently using the ellipsoidal height as an approximation to the orthometric (or geoid) height due to lack of a geoid model.